overlord/devicestate/devicemgr.go

```diff
@@ -1066,6 +1066,58 @@ func (m *DeviceManager) ensureAutoImportAssertions() error {
 	return nil
 }
 
+func (m *DeviceManager) ensureSerialBoundSystemUserAssertionsProcessed() error {
+	if release.OnClassic {
+		return nil
+	}
+
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	var seeded bool
+	if err := m.state.Get("seeded", &seeded); err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+	if !seeded {
+		return nil
+	}
+
+	var waitingOnSerial bool
```

Comment:
Nitpick: wouldn't it make more sense to check this before checking "seeded"? that way we could probably avoid the first call altogether in most cases.

Reply:
yes, thanks